pkg/util/kernel/download_headers_test.go

```diff
@@ -0,0 +1,175 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+// +build linux
+
+package kernel
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/DataDog/nikos/apt"
+	"github.com/DataDog/nikos/types"
+)
+
+const reposSource = "%s/sources.list"
+const reposSourceDir = "%s/sources.list.d"
+
+var _ types.Logger = customLogger{}
+
+var ubuntuRelease = map[string]string{
+	"ID":                 "ubuntu",
+	"ID_LIKE":            "debian",
+	"PRETTY_NAME":        "Ubuntu 20.04.3 LTS",
+	"VERSION_ID":         "20.04",
+	"HOME_URL":           "https://www.ubuntu.com/",
+	"SUPPORT_URL":        "https://help.ubuntu.com/",
+	"BUG_REPORT_URL":     "https://bugs.launchpad.net/ubuntu/",
+	"PRIVACY_POLICY_URL": "https://www.ubuntu.com/legal/terms-and-policies/privacy-policy",
+	"VERSION_CODENAME":   "focal",
+	"UBUNTU_CODENAME":    "focal",
+}
+
+var debianRelease = map[string]string{
+	"PRETTY_NAME":      "Debian GNU/Linux 11 (bullseye)",
+	"NAME":             "Debian GNU/Linux",
+	"VERSION_ID":       "11",
+	"VERSION":          "11 (bullseye)",
+	"VERSION_CODENAME": "bullseye",
+	"ID":               "debian",
+	"HOME_URL":         "https://www.debian.org/",
+	"SUPPORT_URL":      "https://www.debian.org/support",
+	"BUG_REPORT_URL":   "https://bugs.debian.org/",
+}
+
+var ubuntuRepos = []string{
+	"deb http://gb.ports.ubuntu.com/ubuntu-ports/ focal-security main restricted",
+}
+
+var debianRepos = []string{
+	"deb http://http.us.debian.org/debian bullseye main",
+	"deb-src http://http.us.debian.org/debian bullseye main",
+	"deb http://security.debian.org/debian-security bullseye-security main",
+	"deb-src http://security.debian.org/debian-security bullseye-security main",
+	"deb http://http.us.debian.org/debian bullseye-updates main",
+	"deb-src http://http.us.debian.org/debian bullseye-updates main",
+}
+
+var targetUbuntu = types.Target{
+	Distro: types.Distro{
+		Display: "ubuntu",
+		Release: "bullseye/sid",
+		Family:  "debian",
+	},
+	OSRelease: ubuntuRelease,
+	Uname: types.Utsname{
+		Kernel:  "5.4.0-92-generic",
+		Machine: "aarch64",
```

Comment:
Will this work if run on `x86_64`?

Reply:
@brycekahle 
Yes, I have tested on x86_64.
Since we generate the backend manually in the test, the real system configuration does not matter.